tests/integration/test_logcollector/test_keep_running/test_keep_running.py

```diff
@@ -0,0 +1,124 @@
+configuration_ids = [f"rotate_{x['location']}_in_{x['log_format']}_format" for x in metadata]
+
+
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def get_local_internal_options():
+    """Get internal configuration."""
+    return local_internal_options
+
+
+@pytest.fixture(scope="module")
+def generate_log_file():
+    """Generate a log file for testing."""
+    file.write_file(log_test_path, '')
+    yield
+    file.remove_file(log_test_path)
+
+
+def test_keep_running(get_local_internal_options, configure_local_internal_options, get_configuration,
+                      configure_environment, generate_log_file, restart_logcollector):
+    """Check if logcollector keeps running once a log is rotated.
+
+    To do this, logcollector is configured to monitor a log file, then data is added to the log and it is rotated.
+    Finally, write data back to the rotated log and check that logcollector continues to monitor it.
+
+    Args:
+        get_local_internal_options (fixture): Get internal configuration.
+        configure_local_internal_options (fixture): Set internal configuration for testing.
+        get_configuration (fixture): Get configurations from the module.
+        configure_environment (fixture): Configure a custom environment for testing.
+        generate_log_file (fixture): Generate a log file for testing.
+        restart_logcollector (fixture): Reset log file and start a new monitor.
+    """
+    config = get_configuration['metadata']
+
+    # Add first line to log
+    message = config['log_line_before']
+    with open(config['location'], 'a') as f:
+        f.write(f"{message}\n")
+
+    # Ensure that the file is being analyzed
+    message = fr"INFO: \(\d*\): Analyzing file: '{log_test_path}'."
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
+                            callback=monitoring.make_callback(pattern=message,
+                                                              prefix=LOG_COLLECTOR_DETECTOR_PREFIX,
+                                                              escape=False))
+
+    # Add another line to log
+    message = config['log_line_before']
+    with open(config['location'], 'a') as f:
+        f.write(f"{message}\n")
+
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING,
+                            callback=monitoring.make_callback(pattern=message,
+                                                              prefix=LOG_COLLECTOR_DETECTOR_PREFIX,
+                                                              escape=False))
+
+    file.truncate_file(config['location'])
```

Comment:
I think we can get two different tests here. To actually rotate a log, you will need to move it to a new file and then re-create a new one. That's totally different from truncating the file.

If you rotate the log, the daemon will detect the file no longer exists and it will remove its entry from a hash table where all the monitored files are stored. Then, once the logfile re-appears, logcollector must detect it, add it again to the hash table and monitor it from the very beginning.

But, what you implemented is a valid use case too. Someone can truncate the file and the daemon must keep monitoring the file.

Reply:
Done in: https://github.com/wazuh/wazuh-qa/pull/1300/commits/a01dc4462069a2ff448c4d9538d4d38741fe9328